code/VPC-network/security-groups.tf

```diff
@@ -0,0 +1,92 @@
+# security groups to access public (front end) and private (back end)
+# public
+resource "aws_security_group" "Wordpress-front-securitygroup" {
+ 	name = "Wordpress-front-securitygroup"
+ 	description = "front end security group - http and https inbound"
+ 	vpc_id = aws_vpc.wordpress-vpc.id
+ 	tags =	{
+ 		Name = "Wordpress-front-securitygroup"
+ 		}
+	ingress {
+		from_port = 80
+		to_port = 80
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
+		ingress {
+		from_port = 443
+		to_port = 443
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
+	ingress {
+		from_port = "22"
+		to_port = "22"
+		protocol = "tcp"
+		cidr_blocks = [var.MYIP]
+		}
+	egress {
+		from_port = 0
+		to_port = 0
+		protocol = "tcp"
+		cidr_blocks = ["0.0.0.0/0"]
+		}
+	egress {
+		from_port = 443
+		to_port = 443
+		protocol = "tcp"
+		cidr_blocks = [var.vpcCIDR]
+		}
```

Comment:
Dummy question, why we need to allow egress only for the VPC? 😬

Reply:
not dummy question. 
it may be redundant with the 
	egress {
		from_port = 0
		to_port = 0
		protocol = "tcp"
		cidr_blocks = ["0.0.0.0/0"]
		}
I will remove it as the above allows all traffic and the Wordpress-back-securitygroup can do the controlling on what gets through here.

Reply:
will also remove:
	egress { 
        	from_port = 3306
        	to_port = 3306
        	protocol = "tcp"
        	cidr_blocks = [var.pri_subnet_cidr_2, var.pri_subnet_cidr_1]
    }	
for same reasons as above.